Drivers/DevicesInterop/GHIElectronics_TinyCLR_Devices_GHIElectronics_TinyCLR_Devices_Adc_Provider_AdcControllerApiWrapper.cpp

```diff
@@ -1,73 +1,104 @@
+    auto provider = (const TinyCLR_Adc_Controller*)TinyCLR_Interop_GetManager(md, FIELD___impl___I);
+    auto arg1 = TinyCLR_Interop_GetArguments(md, 1);
+    auto ret = TinyCLR_Interop_GetReturn(md);
 
+    ret.Data.Numeric->Boolean = provider->IsChannelModeSupported(provider, (TinyCLR_Adc_ChannelMode)arg1.Data.Numeric->U4) ? true : false;
 
-    return TinyCLR_Result::NotImplemented;
+    return TinyCLR_Result::Success;
 }
 
 TinyCLR_Result Interop_GHIElectronics_TinyCLR_Devices_GHIElectronics_TinyCLR_Devices_Adc_Provider_AdcControllerApiWrapper::GetChannelMode___GHIElectronicsTinyCLRDevicesAdcAdcChannelMode(const TinyCLR_Interop_MethodData md) {
+    auto provider = (const TinyCLR_Adc_Controller*)TinyCLR_Interop_GetManager(md, FIELD___impl___I);
+    auto ret = TinyCLR_Interop_GetReturn(md);
 
+    ret.Data.Numeric->I4 = (int32_t)provider->GetChannelMode(provider);
 
-    return TinyCLR_Result::NotImplemented;
+    return TinyCLR_Result::Success;
 }
 
 TinyCLR_Result Interop_GHIElectronics_TinyCLR_Devices_GHIElectronics_TinyCLR_Devices_Adc_Provider_AdcControllerApiWrapper::SetChannelMode___VOID__GHIElectronicsTinyCLRDevicesAdcAdcChannelMode(const TinyCLR_Interop_MethodData md) {
+    auto provider = (const TinyCLR_Adc_Controller*)TinyCLR_Interop_GetManager(md, FIELD___impl___I);
+    auto arg1 = TinyCLR_Interop_GetArguments(md, 1);
 
-
-    return TinyCLR_Result::NotImplemented;
+    return provider->SetChannelMode(provider, (TinyCLR_Adc_ChannelMode)arg1.Data.Numeric->U4);
 }
 
 TinyCLR_Result Interop_GHIElectronics_TinyCLR_Devices_GHIElectronics_TinyCLR_Devices_Adc_Provider_AdcControllerApiWrapper::OpenChannel___VOID__I4(const TinyCLR_Interop_MethodData md) {
+    auto provider = (const TinyCLR_Adc_Controller*)TinyCLR_Interop_GetManager(md, FIELD___impl___I);
+    auto arg1 = TinyCLR_Interop_GetArguments(md, 1);
 
-
-    return TinyCLR_Result::NotImplemented;
+    return provider->OpenChannel(provider, arg1.Data.Numeric->U4);
 }
 
 TinyCLR_Result Interop_GHIElectronics_TinyCLR_Devices_GHIElectronics_TinyCLR_Devices_Adc_Provider_AdcControllerApiWrapper::CloseChannel___VOID__I4(const TinyCLR_Interop_MethodData md) {
+    auto provider = (const TinyCLR_Adc_Controller*)TinyCLR_Interop_GetManager(md, FIELD___impl___I);
+    auto arg1 = TinyCLR_Interop_GetArguments(md, 1);
 
-
-    return TinyCLR_Result::NotImplemented;
+    return provider->CloseChannel(provider, arg1.Data.Numeric->U4);
 }
 
 TinyCLR_Result Interop_GHIElectronics_TinyCLR_Devices_GHIElectronics_TinyCLR_Devices_Adc_Provider_AdcControllerApiWrapper::Read___I4__I4(const TinyCLR_Interop_MethodData md) {
+    auto provider = (const TinyCLR_Adc_Controller*)TinyCLR_Interop_GetManager(md, FIELD___impl___I);
+    auto arg1 = TinyCLR_Interop_GetArguments(md, 1);
+    auto ret = TinyCLR_Interop_GetReturn(md);
+
+    int32_t value = 0;
 
+    auto error = provider->ReadChannel(provider, arg1.Data.Numeric->U4, value);
```

Comment:
You should be able to inline `ret.Data.Numeric->I4` instead of creating the temp variable. You can then return the call directly.